task/generate-olm-bundle/0.1/generate-olm-bundle.yaml

```diff
@@ -0,0 +1,40 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: generate-olm-bundle
+spec:
+  description: Generate an OLM bundle
+  params:
+    - name: input-dir
+      description: Directory to read cluster-ready operator manifests from
+      default: deploy
+    - name: channels
+      description: Comma-separated list of channels the bundle belongs to
+      default: alpha
+    - name: version
+      description: Semantic version of the operator in the generated bundle
+    - name: package-name
+      description: Bundle's package name
+  workspaces:
+    - name: source
+      description: Workspace with the source code
+  steps:
+    - name: generate-olm-bundle
+      # TODO: ensure this image is built within konflux
+      # TODO: can this be configurable as a parameter? it should be specific to
+      # the target version
+      image: "registry.redhat.io/openshift4/ose-operator-sdk-rhel9:v4.16"
+      workingDir: $(workspaces.source.path)/source
+      args:
+        - generate
+        - bundle
+        - --overwrite
+        - --channels
+        - $(params.channels)
+        - --input-dir
+        - $(params.input-dir)
+        - --version
+        - $(params.version)
+        - --package
+        - $(params.package-name)
```

Comment:
I understand that you are trying to build a task that solves a specific use case but this feels too specific as the task is worded in a way that it should be general purpose.

Reply:
> This new task is highly specific to the way that SRE-P builds and deploys
operators, so it may not be generally useful to others without additional
changes. I'm more than open to a better name for the task or any additional
context needed to ensure it isn't misused or leads to confusion when someone
else is onboarding a standard OLM operator.

:smile: I don't know of a usecase outside of how we are using it, I'm open to any recommendations on how to generalize it more so it is useful for others.